editor/src/messages/tool/tool_messages/spline_tool.rs

```diff
@@ -534,3 +534,102 @@ fn delete_preview(tool_data: &mut SplineToolData, responses: &mut VecDeque<Messa
 	tool_data.preview_point = None;
 	tool_data.preview_segment = None;
 }
+#[cfg(test)]
+mod test_spline_tool {
+	use crate::messages::input_mapper::utility_types::input_mouse::{EditorMouseState, ScrollDelta};
+	use crate::messages::portfolio::document::graph_operation::utility_types::TransformIn;
+	use crate::messages::tool::tool_messages::spline_tool::find_spline;
+	use crate::test_utils::test_prelude::*;
+	use glam::DAffine2;
+
+	#[tokio::test]
+	async fn test_spline_tool_with_transformed_artboard() {
+		let mut editor = EditorTestUtils::create();
+		editor.new_document().await;
+
+		editor.drag_tool(ToolType::Artboard, 0., 0., 500., 500., ModifierKeys::empty()).await;
+
+		let document = editor.active_document();
+		let artboard_layer = document.network_interface.selected_nodes().selected_layers(document.metadata()).next().unwrap();
+
+		editor
+			.handle_message(GraphOperationMessage::TransformSet {
+				layer: artboard_layer,
+				transform: DAffine2::from_scale_angle_translation(DVec2::new(1.5, 1.2), 30.0_f64.to_radians(), DVec2::new(50.0, 25.0)),
+				transform_in: TransformIn::Local,
+				skip_rerender: false,
+			})
+			.await;
+
+		let spline_start = DVec2::new(100., 100.);
+		let spline_mid = DVec2::new(200., 150.);
+		let spline_end = DVec2::new(300., 100.);
+
+		editor.select_tool(ToolType::Spline).await;
+		editor.move_mouse(spline_start.x, spline_start.y, ModifierKeys::empty(), MouseKeys::empty()).await;
+		editor.left_mousedown(spline_start.x, spline_start.y, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: spline_start,
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
```

Comment:
Consider using `editor.click_tool(…)` to simplify the logic. You are also free to write your own helper functions e.g. on `EditorTestUtils`.